internal/types/document.go

```diff
@@ -387,6 +363,34 @@ func (d *Document) isKeyDuplicate(targetKey string) bool {
 	return false
 }
 
+// SetID sets the _id field of the document at the first position.
+func (d *Document) SetID() {
+	if !d.Has("_id") {
```

Comment:
I thought this function sets ID for any document, but I think what it does is it moves ID to the first index. And if a document doesn't have ID, it just does nothing. I suggest renaming it. ReorderIDIfExists or something that is more explicit

Reply:
Renamed it in #1544.